src/main/java/build/buf/protovalidate/EvaluatorBuilder.java

```diff
@@ -76,12 +74,13 @@ class EvaluatorBuilder {
   EvaluatorBuilder(Env env, Config config, List<Descriptor> descriptors, boolean disableLazy)
       throws CompilationException {
     this.env = env;
-    this.descriptors = descriptors;
     this.disableLazy = disableLazy;
     this.rules = new RuleCache(env, config);
 
-    for (Descriptor descriptor : this.descriptors) {
-      this.build(descriptor);
+    if (descriptors != null) {
```

Comment:
The nil check isn't needed here - if anything we could add an `Objects.requireNonNull(descriptors)` above if we wanted with an error message. Doesn't hurt but typically you never pass around null collections.

Reply:
Without the check, it throws an NPE though:  `java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()"`. I think we should do _something_ here to catch that. Seems like an easy thing to prevent.

Reply:
This would only happen though if someone passes `null` to the `buildDescriptors` method, right?

Typically if you wanted a nicer error you'd do `Objects.requireNonNull(descriptors, "descriptors must not be null")` as the first line in the constructor (or the equivalent Preconditions.checkNotNull from Guava).